docs(clp-package): Rewrite S3 log compression guide to reflect new API and script features. (#1510)
y-scope/clp: docs/src/user-docs/guides-using-object-storage/clp-usage.md

## Compressing logs from S3

You can use `sbin/compress-from-s3.sh` to compress logs from S3. It has two modes of operation:

* `s3-object` : Compress S3 objects specified by their full S3 URLs.

> * `s3-key-prefix` : Compress all S3 objects under a given S3 key prefix.

Reviewer: ```suggestion
* [**`s3-object` mode**](#s3-object-compression-mode): Compress S3 objects specified by their full
  S3 URLs.
* [**`s3-key-prefix` mode**](#s3-key-prefix-compression-mode): Compress all S3 objects under a given
  S3 key prefix.
```
I think it'd be nice to include links to the corresponding sections in the document

### `s3-object` compression mode

`s3-object` mode allows you to specify individual S3 objects to compress by their full URLs. To
use this mode, call the `sbin/compress-from-s3.sh` script as follows, replacing the fields in angle

> brackets (`<>`) with the appropriate values:

Reviewer: ```suggestion
use this mode, call the `sbin/compress-from-s3.sh` script as follows, and replace the fields in
angle brackets (`<>`) with the appropriate values:
```
directness

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-object \
  <object-url> [<object-url> ...]
```

* `<object-url>` is a URL identifying the S3 object to compress. It can have one of two formats:
  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<object-key>`
  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<object-key>`

> * The fields in `<object-url>` are as follows:

Reviewer: ```suggestion
  * The fields in `<object-url>` are as follows:
```
some of this is outside the diff range, but this whole point should probably be nested under the `<object-url>` point

Author: This is to follow what we have in the current doc: https://docs.yscope.com/clp/main/user-docs/guides-using-object-storage/clp-usage.html#compressing-logs-from-s3
I'm ok to switch to inner though

Reviewer: fair enough!

  * `<bucket-name>` is the name of the S3 bucket containing your logs.
  * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
    logs.
  * `<object-key>` is the [object key][aws-s3-object-key] of the log file object you wish to
    compress. There must be no duplicate object keys across all `<object-url>` arguments.

Optionally, you can specify the input object URLs through a text file by using the `--inputs-from`
option as follows:

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-object \
  --inputs-from <input-file>
```

* `<input-file>` is a path to a text file containing one S3 object URL **per line**. The URLs must
  follow the same format as described above for `<object-url>`.

:::{note}
`s3-object` mode requires the input object keys to share a non-empty common prefix. If the input
object keys do not share a common prefix, they will be rejected and no compression job will be

> created. This limitation will be relaxed in a future release.

Reviewer: ```suggestion
created. This limitation will be addressed in a future release.
```
best to be vague about what we're gonna do in the future I think

:::

### `s3-key-prefix` compression mode

`s3-key-prefix` mode allows you to compress all objects under a given S3 key-prefix. To use this
mode, call the `sbin/compress-from-s3.sh` script as follows, replacing the fields in angle brackets
(`<>`) with the appropriate values:

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-key-prefix \
  <key-prefix-url>
```

* `<key-prefix-url>` is a URL identifying the S3 key-prefix to compress. It can have one of two
  formats:
  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<key-prefix>`
  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<key-prefix>`
* The fields in `<key-prefix-url>` are as follows:
  * `<bucket-name>` is the name of the S3 bucket containing your logs.
  * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
    logs.
  * `<key-prefix>` is the prefix of all logs you wish to compress and must begin with the

>     `<all-logs-prefix>` value from the [compression IAM policy][compression-iam-policy].

Reviewer: ```suggestion
  * The fields in `<key-prefix-url>` are as follows:
    * `<bucket-name>` is the name of the S3 bucket containing your logs.
    * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
      logs.
    * `<key-prefix>` is the prefix of all logs you wish to compress and must begin with the
      `<all-logs-prefix>` value from the [compression IAM policy][compression-iam-policy].
```
this whole point should probably be nested under the `<key-prefix-url>` point

:::{note}
`s3-key-prefix` mode only accepts a single `<key-prefix-url>` argument. This limitation will be
relaxed in a future release.
:::

> [add-iam-policy]: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_manage-attach-detach.html#embed-inline-policy-console

Reviewer: it's out of the diff range, but this link isn't used anywhere in the document.

Author: True. Let me check.

## Compressing logs from S3

To compress logs from S3, use the `sbin/compress-from-s3.sh` script. The script supports two modes

> of operation. Select a mode by using the corresponding argument in the command:

Reviewer 2: ```suggestion
of operation:
```

  S3 URLs.
* [**s3-key-prefix** mode](#s3-key-prefix-compression-mode): Compress all S3 objects under a given
  S3 key prefix.

### `s3-object` compression mode

`s3-object` mode allows you to specify individual S3 objects to compress by their full URLs. To
use this mode, call the `sbin/compress-from-s3.sh` script as follows, and replace the fields in
angle brackets (`<>`) with the appropriate values:

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-object \
  <object-url> [<object-url> ...]
```

* `<object-url>` is a URL identifying the S3 object to compress. It can be written in either of two
  formats:
  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<object-key>`
  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<object-key>`
* The fields in `<object-url>` are as follows:
  * `<bucket-name>` is the name of the S3 bucket containing your logs.
  * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
    logs.
  * `<object-key>` is the [object key][aws-s3-object-key] of the log file object you wish to

>     compress. There must be no duplicate object keys across all `<object-url>` arguments.

Reviewer 2: ```suggestion
    compress.

    :::{warning}
    There must be no duplicate object keys across all `<object-url>` arguments.
    :::

```

> Instead of typing the input object URLs explicitly in the command, you may specify them in a text

Reviewer 2: ```suggestion
* For a description of other fields, see the [clp-json quick-start
  guide](../quick-start/clp-json.md#compressing-json-logs).

Instead of specifying input object URLs explicitly in the command, you may specify them in a text
```

file and then pass the file into the command using the `--inputs-from` flag, like so:

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-object \
  --inputs-from <input-file>
```

* `<input-file>` is a path to a text file containing one S3 object URL **per line**. The URLs must
  follow the same format as described above for `<object-url>`.

> 

Reviewer 2: ```suggestion
```

### `s3-key-prefix` compression mode

`s3-key-prefix` mode allows you to compress all objects under a given S3 key prefix. To use this
mode, call the `sbin/compress-from-s3.sh` script as follows, and replace the fields in angle
brackets (`<>`) with the appropriate values:

```bash
sbin/compress-from-s3.sh \
  --timestamp-key <timestamp-key> \
  --dataset <dataset-name> \
  s3-key-prefix \
  <key-prefix-url>
```

* `<key-prefix-url>` is a URL identifying the S3 key prefix to compress. It can be written in either
  of two formats:
  * `https://<bucket-name>.s3.<region-code>.amazonaws.com/<key-prefix>`
  * `https://s3.<region-code>.amazonaws.com/<bucket-name>/<key-prefix>`
* The fields in `<key-prefix-url>` are as follows:
  * `<bucket-name>` is the name of the S3 bucket containing your logs.
  * `<region-code>` is the AWS region [code][aws-region-codes] for the S3 bucket containing your
    logs.
  * `<key-prefix>` is the prefix of all logs you wish to compress and must begin with the
    `<all-logs-prefix>` value from the [compression IAM policy][compression-iam-policy].

:::{note}
`s3-key-prefix` mode accepts a single `<key-prefix-url>` argument. This limitation will be addressed

> in a future release.

Reviewer 2: ```suggestion
`s3-key-prefix` mode only accepts a single `<key-prefix-url>` argument. This limitation will be
addressed in a future release.
```

:::

[aws-s3-object-key]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

> [aws-region-codes]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Availability

Reviewer 2: ```suggestion
[aws-region-codes]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Availability
[aws-s3-object-key]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
```

To alphabetize.
